backend/src/core/services/auth.service.ts

```diff
@@ -224,7 +221,7 @@ export const InitAuthService = (redisService: RedisService): AuthService => {
     await saveHashedOtp(email, hashedOtp)
 
     const appName = config.get('APP_NAME')
-    return MailService.mailClient.sendMail({
+    await MailService.mailClient.sendMail({
```

Comment:
Hmm, I'm a little confused about the `void` operator. It seems like an esoteric feature of JS that isn't specific to calling async functions where we don't care about the return value? It also seems that Node.js may prematurely terminate async functions that aren't explicitly being awaited?
https://stackoverflow.com/questions/46966890/what-happens-when-a-promise-never-resolves

Reply:
So the NodeJS event loop is a little too complicated to address here ([this](https://www.youtube.com/watch?v=8aGhZQkoFbQ) is the go-to explainer). 

You're right to say that JS's `void` operator (which is distinct from the `void` return type in TS) has a lot of weird history that tbh I'm not super familiar wtih. 

For our purposes, because `sendMail` itself does the error handling and we're not using the return value in this function, it's OK to use `void`. See example discussions [here](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-floating-promises.md), [here](https://github.com/Microsoft/TypeScript/issues/13376#issuecomment-273289748), and [here](https://github.com/palantir/tslint/issues/4653). We have also used this in the codebase previously to fulfill this linting rule, and things seem to work fine!